nibabel/cifti2/cifti2.py

```diff
@@ -88,6 +88,23 @@ class Cifti2HeaderError(Exception):
                           'CIFTI_STRUCTURE_THALAMUS_LEFT',
                           'CIFTI_STRUCTURE_THALAMUS_RIGHT')
 
+# "Standard CIFTI Mapping Combinations" within CIFTI-2 spec
+# https://www.nitrc.org/forum/attachment.php?attachid=341&group_id=454&forum_id=1955
+CIFTI_EXTENSIONS_TO_INTENTS = {
+    '.dconn': 'NIFTI_INTENT_CONNECTIVITY_DENSE',
+    '.dtseries': 'NIFTI_INTENT_CONNECTIVITY_DENSE_SERIES',
+    '.pconn': 'NIFTI_INTENT_CONNECTIVITY_PARCELLATED',
+    '.ptseries': 'NIFTI_INTENT_CONNECTIVITY_PARCELLATED_SERIES',
+    '.dscalar': 'NIFTI_INTENT_CONNECTIVITY_DENSE_SCALARS',
+    '.dlabel': 'NIFTI_INTENT_CONNECTIVITY_DENSE_LABELS',
+    '.pscalar': 'NIFTI_INTENT_CONNECTIVITY_PARCELLATED_SCALAR',
+    '.pdconn': 'NIFTI_INTENT_CONNECTIVITY_PARCELLATED_DENSE',
+    '.dpconn': 'NIFTI_INTENT_CONNECTIVITY_DENSE_PARCELLATED',
+    '.pconnseries': 'NIFTI_INTENT_CONNECTIVITY_PARCELLATED_PARCELLATED_SERIES',
+    '.pconnscalar': 'NIFTI_INTENT_CONNECTIVITY_PARCELLATED_PARCELLATED_SCALAR',
+    '.dfan': 'NIFTI_INTENT_CONNECTIVITY_DENSE_SERIES',
+}
```

Comment:
Does it make sense to add an `extension` field to `nifti1.intent_codes`? 

https://github.com/nipy/nibabel/blob/0ba55d4966cb3b382583b66f7e64edb17df7ce47/nibabel/nifti1.py#L152-L250

The `NIFTI_INTENT_CONNECTIVITY_*` are added here:

https://github.com/nipy/nibabel/blob/0ba55d4966cb3b382583b66f7e64edb17df7ce47/nibabel/cifti2/parse_cifti2.py#L50-L75

Reply:
do you mean within the parameters description tuple? or as a new element? I'm not sure how relevant that would be outside of the CIFTI scope tbh

Reply:
New field. And it would be None for most. GIFTI could use it for its extensions, as well. Though on second thought, since GIFTI has at least one that is a combination of intents, we might want to create a separate structure.

I'll think more about it.

Reply:
I think I agree that this probably shouldn't be shoehorned into the intent code structure.

The extensions do also come with required axis/index types. What do you think of something like the following?

```Python
CIFTI_EXTENSIONS = Recoder((
    ('dconn', 'NIFTI_INTENT_CONNECTIVITY_DENSE', ('CIFTI_INDEX_TYPE_BRAIN_MODELS', 'CIFTI_INDEX_TYPE_BRAIN_MODELS')),
    ('dtseries', 'NIFTI_INTENT_CONNECTIVITY_DENSE_SERIES', ('CIFTI_INDEX_TYPE_SERIES', 'CIFTI_INDEX_TYPE_BRAIN_MODELS')),
    ...
    ), fields=('extension', 'niistring', 'map_types')
)
```

This would set us up to be able to validate or potentially reorder axes, as well.